python/imjoy_rpc/connection/jupyter_connection.py

```diff
@@ -29,9 +30,15 @@ def __init__(self, rpc_context):
         self.interface = None
         self.rpc_context = rpc_context
         self._codecs = {}
-        self.kernel_id = re.search(
-            "kernel-(.*).json", ipykernel.connect.get_connection_file()
-        ).group(1)
+        connection_file = ipykernel.connect.get_connection_file()
+        if 'kernel-' in connection_file:
+            self.kernel_id = re.search(
+                "kernel-(.*).json", ipykernel.connect.get_connection_file()
+            ).group(1)
+        else:
+            # nbmake
+            self.kernel_id = Path(connection_file).stem
+        print('kernel_id', self.kernel_id)
```

Comment:
Do we want to keep the print statement here?

Reply:
ah, no! :-)